pkg/shared/families/infofinder/sshtopology/scanner.go

```diff
@@ -0,0 +1,364 @@
+			// Maybe we need to support to send both errors and findings in a higher level.
+			s.logger.Error(retResults.Error)
+			retResults.Error = nil
+		}
+
+		s.sendResults(retResults, nil)
+	}()
+
+	return nil
+}
+
+func getHomeUserDirs(rootDir string) ([]string, error) {
+	var dirs []string
+	homeDirPath := path.Join(rootDir, "home")
+	files, err := os.ReadDir(homeDirPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read dir (%v): %v", homeDirPath, err)
+	}
+
+	for _, f := range files {
+		if f.IsDir() {
+			dirs = append(dirs, path.Join(homeDirPath, f.Name()))
+		}
+	}
+
+	return dirs, nil
+}
+
+func (s *Scanner) getSSHDaemonKeysFingerprints(rootPath string) ([]types.Info, error) {
+	paths, err := s.getPrivateKeysPaths(path.Join(rootPath, "/etc/ssh"), false)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get private keys paths: %v", err)
+	}
+	s.logger.Debugf("Found ssh daemon private keys paths %+v", paths)
+
+	fingerprints, err := s.getFingerprints(paths, types.SSHDaemonKeyFingerprint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ssh daemon private keys fingerprints: %v", err)
+	}
+	s.logger.Debugf("Found ssh daemon private keys fingerprints %+v", fingerprints)
+
+	return fingerprints, nil
+}
+
+func (s *Scanner) getSSHPrivateKeysFingerprints(homeUserDir string) ([]types.Info, error) {
+	paths, err := s.getPrivateKeysPaths(homeUserDir, true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get private keys paths: %v", err)
+	}
+	s.logger.Debugf("Found ssh private keys paths %+v", paths)
+
+	infos, err := s.getFingerprints(paths, types.SSHPrivateKeyFingerprint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ssh private keys fingerprints: %v", err)
+	}
+	s.logger.Debugf("Found ssh private keys fingerprints %+v", infos)
+
+	return infos, nil
+}
+
+func (s *Scanner) getSSHAuthorizedKeysFingerprints(homeUserDir string) ([]types.Info, error) {
+	infos, err := s.getFingerprints([]string{path.Join(homeUserDir, ".ssh/authorized_keys")}, types.SSHAuthorizedKeyFingerprint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ssh authorized keys fingerprints: %v", err)
+	}
+	s.logger.Debugf("Found ssh authorized keys fingerprints %+v", infos)
+
+	return infos, nil
+}
+
+func (s *Scanner) getSSHKnownHostsFingerprints(homeUserDir string) ([]types.Info, error) {
+	infos, err := s.getFingerprints([]string{path.Join(homeUserDir, ".ssh/known_hosts")}, types.SSHKnownHostFingerprint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ssh known hosts fingerprints: %v", err)
+	}
+	s.logger.Debugf("Found ssh known hosts fingerprints %+v", infos)
+
+	return infos, nil
+}
+
+func (s *Scanner) getFingerprints(paths []string, infoType types.InfoType) ([]types.Info, error) {
+	var infos []types.Info
+
+	for _, p := range paths {
+		_, err := os.Stat(p)
+		if os.IsNotExist(err) {
+			s.logger.Debugf("File (%v) does not exist.", p)
```

Comment:
Why is it not treated as error?

Reply:
If a file is missing I don't want to spam the logs.
For example, if `.ssh/authorized_keys` is missing we don't care.